src/dataServices/dynamoDbUtil.ts

```diff
@@ -29,10 +32,20 @@ export class DynamoDbUtil {
         return cleanedItem;
     }
 
-    static prepItemForDdbInsert(resource: any, id: string, vid: number, documentStatus: DOCUMENT_STATUS) {
+    static prepItemForDdbInsert(
+        resource: any,
+        id: string,
+        vid: number,
+        documentStatus: DOCUMENT_STATUS,
+        ttlInSeconds?: number,
+    ) {
         const item = clone(resource);
         item.id = id;
         item.vid = vid;
+        if (!_.isUndefined(ttlInSeconds)) {
+            const unixNow: number = Math.floor(Date.now() / 1000);
+            item[TTL_IN_SECONDS] = unixNow + ttlInSeconds;

```

Comment:
interesting so this is more of expiration time than TTL?

Reply:
The dynamo DB documentation defines it as a TTL so I tried to follow suit. When people pass the map to the constructor it is actually a ttl but dynamodb expects an expiry date.